tests/testthat/_snaps/eval-relocate.md

```diff
@@ -0,0 +1,109 @@
+# can't supply both `before` and `after`
+
+    Code
+      relocate_loc(c(x = 1), before = 1, after = 1)
+    Condition
+      Error in `relocate_loc()`:
+      ! Can't supply both `before` and `after`.
+
+---
+
+    Code
+      relocate_loc(c(x = 1), before = 1, after = 1, before_arg = ".before",
+      after_arg = ".after")
+    Condition
+      Error in `relocate_loc()`:
+      ! Can't supply both `.before` and `.after`.
+
+# can't relocate with out-of-bounds variables by default
+
+    Code
+      (expect_error(relocate_loc(x, c)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
+    Code
+      (expect_error(relocate_loc(x, a, before = c)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
+    Code
+      (expect_error(relocate_loc(x, a, after = c)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
```

Comment:
These aren't the best errors, but I don't see an easy way to improve them.

Possible improvements:
- `Can't relocate` vs `Can't subset`
- Mention if the problem was in `expr` vs `before` vs `after`

Reply:
I have reopened the vctrs and tidyselect issues for this, it seems worth it now